pkg/schemas/service.go

```diff
@@ -223,6 +227,67 @@ func (s *Service) DeleteSchema(ctx context.Context, ID string) error {
 
 }
 
+func (s *Service) GetDefaultSchema(ctx context.Context) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.GetDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	ID, ok := cm.Data[DEFAULT_SCHEMA]
+
+	if !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
+
+	defaultSchema := new(DefaultSchema)
+	defaultSchema.ID = ID
+
+	return defaultSchema, nil
+}
+
+func (s *Service) UpdateDefaultSchema(ctx context.Context, schema *DefaultSchema) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.UpdateDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	if _, ok := cm.Data[DEFAULT_SCHEMA]; !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
+
+	//This may be unnecessary, but my understanding is that the default.schema field is either supposed to be default or an identity schema that is included in the configmap
+	// this section checks if the input ID has a corresponding schema
+	schemaMap := make(map[string]bool)
+	for k := range cm.Data {
+		if k == DEFAULT_SCHEMA {
+			continue
+		}
+		schemaMap[k] = true
+	}
+	schemaMap[cm.Data[DEFAULT_SCHEMA]] = true
+	if _, ok := schemaMap[schema.ID]; !ok {
+		return nil, fmt.Errorf("schema with ID %s not available", schema.ID)
+	}
```

Comment:
Not sure why this is needed. From what I can tell: you are copying the keys of a map to a set, then you also add the default schema ID in the set and then you check if the new default schema is in the set.

Why can't you just check if the original map contains the new default schema ID?

Reply:
I can see that this is too much. The idea was to make sure that the new default schema ID is:
- either in the original map,
- or the default value of "default.schema" key in the original map for API to be idempotent
-  but the new schema ID can't be "default.schema"

I think these conditions should still be enforced.

Reply:
I don't think we should care if the value of the `default.schema` changes, it makes no difference. 
Being idempotent means that even if the user performs the same action many times, the result will be the same. With the current logic, if the user updates the default schema they will get a `200` response, but if they try to make the same call again they will get a `500`.

Reply:
I see your point. I modified the code.